src/log.jl

```diff
@@ -178,24 +182,26 @@ end
 Compute the natural logarithm of `x`. The inverse of the natural logarithm is
 the natural expoenential function `exp(x)`
 """
-function log_fast(d::T) where {T<:Union{Float32,Float64}}
+function log_fast(d::FloatType)
+    T = eltype(d)
     o = d < floatmin(T)
-    o && (d *= T(Int64(1) << 32) * T(Int64(1) << 32))
+    d = vifelse(o, d * T(Int64(1) << 32) * T(Int64(1) << 32), d)
 
     e = ilogb2k(d * T(1.0/0.75))
     m = ldexp3k(d, -e)
-    o && (e -= 64)
+    e = vifelse(o, e - 64, e)
 
     x  = (m - 1) / (m + 1)
     x2 = x * x
 
     t = log_fast_kernel(x2)
-    
-    x = x * t + T(MLN2) * e
-    
-    isinf(d) && (x = T(Inf))
-    (d < 0 || isnan(d)) && (x = T(NaN))
-    d == 0 && (x = -T(Inf))
+
+    x = muladd(x, t, T(MLN2) * e)
```

Comment:
I'm not sure you can safely replace the previous code with a muladd, I recall this replacement actually makes it less accurate missing the  ulp requirements.

Reply:
Yes let's try to keep the PR as minimal, and we can open up further PRs if we see that the accuracy is not modified, although I'm pretty sure it is, since I recall trying this.